src/util/string_expr.h

```diff
@@ -119,29 +114,101 @@ class string_exprt: public struct_exprt
 
   equal_exprt axiom_for_has_length(const exprt &rhs) const
   {
+    PRECONDITION(rhs.type()==length().type());
     return equal_exprt(length(), rhs);
   }
 
   equal_exprt axiom_for_has_length(mp_integer i) const
   {
     return axiom_for_has_length(from_integer(i, length().type()));
   }
+};
+
+// Representation of strings as arrays
+class array_string_exprt:
+    public string_exprt<array_string_exprt>,
+    public exprt
+{
+public:
+  exprt &length()
+  {
+    return to_array_type(type()).size();
+  }
+
+  const exprt &length() const
+  {
+    return to_array_type(type()).size();
+  }
+
+  exprt &content()
+  {
+    return *this;
+  }
+
+  const exprt &content() const
+  {
+    return *this;
+  }
+};
+
+inline array_string_exprt &to_array_string_expr(exprt &expr)
+{
+  PRECONDITION(expr.type().id()==ID_array);
```

Comment:
No there is no specific id because I don't want to change the original id there.
I just want a class for expressions that have type array.
An additional check we could have here is that the subtype is a (un)signedbv type.